src/main/scala/mesosphere/mesos/ResourceMatcher.scala

```diff
@@ -1,42 +1,83 @@
 package mesosphere.mesos
 
 import mesosphere.marathon.core.launcher.impl.TaskLabels
-import mesosphere.marathon.core.task.Task
-import mesosphere.marathon.state.{ PersistentVolume, ResourceRole, RunSpec, DiskType, DiskSource }
-import mesosphere.marathon.tasks.{ PortsMatch, PortsMatcher }
-import mesosphere.mesos.protos.Resource
+import mesosphere.marathon.state.{ PersistentVolume, ResourceRole, DiskSource }
 import org.apache.mesos.Protos
-import org.apache.mesos.Protos.Resource.DiskInfo.Source
 import org.apache.mesos.Protos.Offer
 import org.slf4j.LoggerFactory
 
 import scala.annotation.tailrec
 import scala.collection.JavaConverters._
 import scala.collection.immutable.Seq
-import scala.collection.immutable
 import scala.collection.mutable
+import scala.collection
+
+trait ResourceMatcher extends ((String, Iterable[Protos.Resource]) => Iterable[MatchResult]) {
+  val resourceName: ResourceMatcher.Role
```

Comment:
this is confusing. the name of a resource is distinct from the resource role. making `resourceName` of type `Role` doesn't jive. if this is actually a role name field, please rename to `roleName`.